.github/workflows/rune_test.yml

```diff
@@ -0,0 +1,147 @@
+name: Rune Test
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+on: [push, pull_request]
```

Comment:
We could handle the push request only. Two reasons: 1. Security: we are not able to control who could send a pull request, so any code would execute on this machine. 2. Bandwidth: only one HW may not able to handle all the pull requests.